src/components/common/Select.ts

```diff
@@ -165,47 +167,6 @@ export type SelectProps<
   options: (Value | Option)[];
 };
 
```

Comment:
I think the other things related to Select should also be moved, like `SelectProps`, `BaseSelect`, etc. Why not move the entire contents of `Select.ts`?

Reply:
There are couple of reason:
1) Workbench needs just Select
2) Since this is the first time i was extracting the component, i wanted to do it one by one, Once this PR is merged, i will have a follow up PR that exports baseslect/selectProps etc from common folder

Reply:
For what it's worth, `GroupedSelect` (and with it, some of the shared pieces like `BaseSelect`) looks fairly straightforward to move. I could see adding it to this PR or moving it in a follow on.

`VirtualizedMenuList` and `VirtualizedSelect` are tied into Terra UI testing because `VirtualizedSelectAutoSizer` gets automatically mocked, so I'm definitely supportive of leaving _those_ for another PR.